### PR TITLE
[READY] [Please merge me] Fermichem tweaks: Purity, Astral, RNG, MK, SMilk, SDGF, furranium, hatmium, eigen, nanite.

### PR DESCRIPTION
**Pull request by Thalpy**
## About The Pull Request

Changes the way a few chems work, and fixes a few problems

**General** Reworks purity to be cached, so that purity calculations don't compound on mob to mob transfers.

**Astral:** Added a few things to reduce the chance that procs will be called at the same time - causing possession mechanics to wig out and cause players to be stuck in each other's bodies.

**randomchem:** Now allows people with enough dedication to cause the effects permanently upon crit players, counting for the assassination objective, without actually killing the player. Finally, a non lethal method! Be careful to not leave any lying around, as spraying the effected players with the chem will revert them back.

**MKUltra:** Tweaks language, adds a few british phrases and adds a few more pref checks on it's use.

**SMilk:** Rate of milk regen now scales from the base regen, with respect to size. (request from community & cebu) 

**SDGF:** Toxin now processes slower, due to the difficult of creation, and the healing chem now processes slightly slower too, to reduce the chance of clones dying on creation. 

**Furanium** Reduces the strictness on the purity so that you can slightly drift from optimal (i.e. so that it accepts 0.9999999999999).

**Hatmium:** Hat armour growth now relies on purity, and hats will now get angry if they're not on a head. Lets you create an armourless hat too by spraying the ground.

**Eigenstasium** fixes a bug that allowed impure chem teleportation, and prevents return teleport if the mob has stabilizing agent inside them.

**Nanite bane** buffed removal rate so that you don't need to take 300u to get rid of healthy nanites, and made rate equal to purity. You still need a substantial amount to make it, and considering it's intense difficulty, I think this is warranted (I don't think you could make 300u unless you spent the whole round trying).

## Why It's Good For The Game

Tweaks and fixes! Hopefully makes things a bit easier to work too.

## Changelog
:cl: Fermis
tweak: Purity, Astral, RNG, MK, SMilk, SDGF, furranium, hatmium, eigen, nanite.
fix: Eigen and purity.
/:cl:

### CODE REVIEW

**Comment by MrJWhit** [comment]
Politic-aid should absolutely be it's own separate PR, it completely changes how the game works, and it's not a "tweak and fix".

**Comment by Thalpy** [comment]
Actually I don't need comments to do that.

**Comment by MrJWhit** [comment]
But... it's still not a minor thing. It can **remove antag** statuses, No other thing can do that, not even borging removes the antag datum. It should be it's own PR that everyone can talk about, without interfering with these other changes.

**Comment by Thalpy** [comment]
> But... it's still not a minor thing. It can **remove antag** statuses, No other thing can do that, not even borging removes the antag datum. It should be it's own PR that everyone can talk about, without interfering with these other changes.

I might be able to find a way actually.

**Comment by MrJWhit** [comment]
If you find a way to remove the antag datum, it should be patched out, because it's not intended.

**Comment by Thalpy** [comment]
> If you find a way to remove the antag datum, it should be patched out, because it's not intended.

I meant to atomise it easily, I'm just not thinking straight.

**Comment by MrJWhit** [comment]
All right, this just feels deeply underhanded with how this is titled and projected as.

**Comment by Thalpy** [comment]
> All right, this just feels deeply underhanded with how this is titled and projected as.

I had mentioned it before in the discord, but I can see what you mean, and I wrote out a list of every change in detail.

**Comment by Scorpion-117** [comment]
I'm just going to say make that politic chem a seperate PR. It's a chem that messes with antags on a deconversion level and isn't something that can just be part of a PR Fix. If that makes sense

**Comment by Thalpy** [comment]
I've removed it.

**Comment by MrJWhit** [comment]
It still exists in recipes/fermi.dm and reagents/MKUltra.dm
The chemical itself is still there and the ability to create it, it's just that it doesn't have the antag removing status. Which, adding a new chem is still fine, it just should be it's on PR.

**Comment by Thalpy** [comment]
I meant the antag datum stuff.

**Comment by deathride58** [comment]
The ability to remove antag statuses of non-conversion antags is a hard no, regardless of how hard the chem is to make. Removing antag statuses of non-conversion antags causes numerous issues ranging from making it much harder to tell if someone is a griefer or an antag that just got deconverted, the inability for players to tell who was actually antag since it removes their spot on the roundend display, and more.

**Comment by Thalpy** [comment]
Well, I don't have to worry about putting it back in at least.

**Comment by Thalpy** [comment]
Just give me a sec to change stuff, I had thought I'd be able to write a doc before being jumped on, and I've another PR I atomised out that I want to get off my mind.

**Comment by Thalpy** [comment]
Okay now it's fully removed.

File: code/modules/mob/living/simple_animal/astral.dm [hunk 1]
```diff
@@ -57,3 +57,9 @@
 				return
 			to_chat(A, "[src] projects into your mind, <b><i> \"[message]\"</b></i>")
 			log_game("FERMICHEM: [src] has astrally transmitted [message] into [A]")
+
+//Delete the mob if there's no mind! Pay that mob no mind.
+/mob/living/simple_animal/astral/Life()
+	if(!src.mind)
```
Comment (silicons) [inline code/modules/mob/living/simple_animal/astral.dm]: no src.

Comment (Thalpy) [reply code/modules/mob/living/simple_animal/astral.dm]: Fixed

File: code/modules/mob/living/simple_animal/astral.dm [hunk 2]
```diff
@@ -57,3 +57,9 @@
 				return
 			to_chat(A, "[src] projects into your mind, <b><i> \"[message]\"</b></i>")
 			log_game("FERMICHEM: [src] has astrally transmitted [message] into [A]")
+
+//Delete the mob if there's no mind! Pay that mob no mind.
+/mob/living/simple_animal/astral/Life()
+	if(!mind)
+		qdel(src)
+	..()
```
Comment (deathride58) [inline code/modules/mob/living/simple_animal/astral.dm]: . = ..()

File: code/modules/surgery/organs/vocal_cords.dm [hunk 3]
```diff
@@ -1082,15 +1082,15 @@
 			switch(E.phase)
 				if(1 to 2)
 					E.phase = -1
-					to_chat(C, "<span class='big warning'>You have no recollection of being enthralled by [E.master]!</b></span>")
+					to_chat(C, "<span class='big warning'>You suddenly forget [E.master]'s modulus operandi and their political affiliations!</b></span>")
```
Comment (deathride58) [inline code/modules/surgery/organs/vocal_cords.dm]: This isn't really too clear from the enthrallee's point of view.  The former flavortext was a lot more clear of exactly what is supposed to happen.

File: modular_citadel/code/datums/status_effects/chems.dm [hunk 4]
```diff
@@ -710,15 +741,6 @@
...
 		deltaResist *= 1.5
-	if (!HAS_TRAIT(owner, TRAIT_CROCRIN_IMMUNE) && M.canbearoused)
-		if (owner.reagents.has_reagent("anaphro"))
-			deltaResist *= 1.5
-		if (owner.reagents.has_reagent("anaphro+"))
-			deltaResist *= 2
-		if (owner.reagents.has_reagent("aphro"))
-			deltaResist *= 0.75
-		if (owner.reagents.has_reagent("aphro+"))
-			deltaResist *= 0.5
```
Comment (deathride58) [inline modular_citadel/code/datums/status_effects/chems.dm]: Is this really necessary to remove? It'd probably be better to add a lewdchem check like the other things that had lewdchem checks added

File: modular_citadel/code/modules/reagents/chemistry/reagents/fermi_reagents.dm [hunk 5]
```diff
@@ -58,15 +54,20 @@
 		return ..()
 	var/hatArmor = 0
 	if(!overdosed)
-		hatArmor = (purity/10)
+		hatArmor = (cached_purity/10)
 	else
-		hatArmor = - (purity/10)
-	if(hatArmor > 90)
+		hatArmor = (cached_purity/10)
+	if(hatArmor > 100)
```
Comment (deathride58) [inline modular_citadel/code/modules/reagents/chemistry/reagents/fermi_reagents.dm]: 100 being the max armor value is a hard no.

**Review by deathride58 — DISMISSED** [review]

File: code/modules/surgery/organs/vocal_cords.dm (hunk 3 above)
Comment (Thalpy) [reply code/modules/surgery/organs/vocal_cords.dm]: Reverted

File: modular_citadel/code/datums/status_effects/chems.dm (hunk 4 above)
Comment (Thalpy) [reply modular_citadel/code/datums/status_effects/chems.dm]: People saying that MKUltra was made purely for lewds wore me down. I wanted to see the mechanics in action as I intended them, but, I've reverted this and also put in the check.

File: modular_citadel/code/modules/reagents/chemistry/reagents/fermi_reagents.dm (hunk 5 above)
Comment (Thalpy) [reply modular_citadel/code/modules/reagents/chemistry/reagents/fermi_reagents.dm]: I'm not sure how this was the case, its back to 90, I might've coped old code by accident. Reverted.

File: code/modules/mob/living/simple_animal/astral.dm (hunk 2 above)
Comment (Thalpy) [reply code/modules/mob/living/simple_animal/astral.dm]: Fixed.

File: modular_citadel/code/datums/status_effects/chems.dm [hunk 6]
```diff
@@ -161,17 +161,49 @@
...
+		owner.mind.transfer_to(G)//Just in case someone else is inside of you, it makes them a ghost and should hopefully bring them home at the end.
+		to_chat(G, "<span class='warning'>[M]'s conciousness snaps back to them as their astrogen runs out, kicking your projected mind out!'</b></span>")
+		log_game("FERMICHEM: [M]'s possesser has been booted out into a astral ghost!")
+	originalmind.transfer_to(original)
+
+
 
 /*//////////////////////////////////////////
-		Mind control functions
+		Political functions!
```
Comment (silicons) [inline modular_citadel/code/datums/status_effects/chems.dm]: ??

File: code/modules/surgery/organs/vocal_cords.dm [hunk 7]
```diff
@@ -1089,7 +1089,7 @@
 					if(C.client?.prefs.lewdchem)
 						addtimer(CALLBACK(GLOBAL_PROC, .proc/to_chat, C, "<span class='big warning'>You revert to yourself before being enthralled by your [E.enthrallGender], with no memory of what happened.</b></span>"), 5)
 					else
-						addtimer(CALLBACK(GLOBAL_PROC, .proc/to_chat, C, "<span class='big warning'>You revert to who you were before, with no memory of what happened with [E.master].</b></span>"), 5)
+						addtimer(CALLBACK(GLOBAL_PROC, .proc/to_chat, C, "<span class='big warning'>You revert to who you were before, with no memory of what happened with [E.master], nor their political affiliations.</b></span>"), 5)
```
Comment (silicons) [inline code/modules/surgery/organs/vocal_cords.dm]: ??

File: code/modules/surgery/organs/vocal_cords.dm [hunk 8]
```diff
@@ -749,45 +749,45 @@
 		to_chat(world, "[user]'s power is [power_multiplier].")
 
 	//Mixables
-	var/static/regex/enthral_words = regex("relax|obey|love|serve|docile|so easy|ara ara")
-	var/static/regex/reward_words = regex("good boy|good girl|good pet|good job")
-	var/static/regex/punish_words = regex("bad boy|bad girl|bad pet|bad job")
+	var/static/regex/enthral_words = regex("relax|obey|love|serve|so easy|ara ara|old boy|pip pip|whatho|how about we discuss this|spot of tea")
```
Comment (silicons) [inline code/modules/surgery/organs/vocal_cords.dm]: intended?

File: modular_citadel/code/modules/reagents/chemistry/reagents/MKUltra.dm [hunk 9]
```diff
@@ -132,7 +132,7 @@ Creating a chem with a low purity will make you permanently fall in love with so
 /datum/reagent/fermi/enthrall
 	name = "MKUltra"
 	id = "enthrall"
-	description = "A forbidden deep red mixture that overwhelms a foreign body with waves of pleasure, intoxicating them into servitude. When taken by the creator, it will enhance the draw of their voice to those affected by it."
+	description = "A forbidden deep red mixture that increases a person's succeptability to another's words. When taken by the creator, it will enhance the draw of their voice to those affected by it."
```
Comment (silicons) [inline modular_citadel/code/modules/reagents/chemistry/reagents/MKUltra.dm:135]: intended?

File: modular_citadel/code/modules/reagents/chemistry/reagents/MKUltra.dm [hunk 10]
```diff
@@ -194,7 +194,7 @@ Creating a chem with a low purity will make you permanently fall in love with so
 			Vc.Remove(M)
 		nVc.Insert(M)
 		qdel(Vc)
-		to_chat(M, "<span class='notice'><i>You feel your vocal chords tingle as your voice comes out in a more sultry tone.</span>")
+		to_chat(M, "<span class='notice'><i>You feel your vocal chords tingle as you speak with a charasmatic tone.)]</span>")
```
Comment (silicons) [inline modular_citadel/code/modules/reagents/chemistry/reagents/MKUltra.dm]: intended?

File: code/modules/surgery/organs/vocal_cords.dm (hunk 8 above)
Comment (Thalpy) [reply code/modules/surgery/organs/vocal_cords.dm]: I had originally split it, I'll revert it.

**Comment by Thalpy** [comment]
HELP

File: modular_citadel/code/datums/status_effects/chems.dm (hunk 6 above)
Comment (Thalpy) [reply modular_citadel/code/datums/status_effects/chems.dm]: fixed

File: modular_citadel/code/modules/reagents/chemistry/reagents/MKUltra.dm (hunk 10 above)
Comment (Thalpy) [reply modular_citadel/code/modules/reagents/chemistry/reagents/MKUltra.dm]: fixed

**Comment by Thalpy** [comment]
Somehow I fixed it. It should be okay now.

File: code/modules/hydroponics/grown/cotton.dm [hunk 11]
```diff
@@ -1,79 +1,79 @@
-/obj/item/seeds/cotton
```
Comment (Putnam3145) [inline code/modules/hydroponics/grown/cotton.dm:1]: how

File: modular_citadel/code/modules/reagents/chemistry/reagents/fermi_reagents.dm [hunk 12]
```diff
@@ -389,20 +393,30 @@
...
+		catto.origin = H
 
 /datum/reagent/fermi/secretcatchem/on_mob_life(mob/living/carbon/H)
+	if(!catto)
+		metabolization_rate = 5
+		return ..()
 	if(catto.health <= 0) //So the dead can't ghost
 		if(prob(10))
-			to_chat(H, "<span class='notice'>You feel your body start to slowly shift back from it's dead form.</span>")
+			to_chat(catto, "<span class='notice'>You feel your body start to slowly shift back from it's dead form.</span>")
```
Comment (Putnam3145) [inline modular_citadel/code/modules/reagents/chemistry/reagents/fermi_reagents.dm:408]: its

**Review by Putnam3145 — APPROVED** [review]

**Pull request merged**